### PR TITLE
Handle all transactions using the transaction wrapper

### PR DESCRIPTION
**Pull request by gammazero**
- Fix intermittent test fail, due to varying ordering of map
- Wrap additional functions in transaction

### CODE REVIEW

File: controller/state/statedb.go [hunk 1]
```diff
@@ -632,27 +647,25 @@ func (s *stateDB) countTasks(ctx context.Context) (int, error) {
...
+		cidLink, err := cid.Decode(c)
+		if err != nil {
+			return err
+		}
 
-	if err := tx.Commit(); err != nil {
+		loader := txContextLoader(ctx, tx)
+		if err = (cidlink.Link{Cid: cidLink}).Load(ctx, ipld.LinkContext{}, na, loader); err != nil {
+			return err
+		}
```
Comment (willscott) [inline controller/state/statedb.go:665]: this can lead to a sub-txn, and should be done outside of this txn, i think?

Comment (hannahhoward) [reply controller/state/statedb.go:665]: I don't think so no? cause the txContextLoader takes the passed in transaction.

File: controller/state/statedb.go [hunk 2]
```diff
@@ -511,26 +519,33 @@ func (s *stateDB) NewRetrievalTask(ctx context.Context, retrievalTask tasks.Retr
...
-		if err = rows.Scan(&status, &stage, &run, &ts); err != nil {
-			return nil, err
+	err := s.transact(ctx, func(tx *sql.Tx) error {
+		rows, err := tx.QueryContext(ctx, taskHistorySQL, taskID)
+		if err != nil {
+			return err
+		}
+		defer rows.Close()
+
+		history = nil
```
Comment (hannahhoward) [inline controller/state/statedb.go:530]: Non-blocking:
I assume this line is cause `transact` may retry -- we should probably indicate somewhere that retries are possible and if you're using closures to access higher scoped vars you better reset them. took me a few rereads to figure out what this mean or why it was needed.

**Review by hannahhoward — APPROVED** [review]
LGTM with a couple comments

File: controller/state/statedb.go (hunk 2 above)
Comment (gammazero) [reply controller/state/statedb.go:530]: Nice deduction!  That's exactly why.  I will add comments.

Comment (gammazero) [reply controller/state/statedb.go]: Done

**Review by willscott — APPROVED** [review]

**Review by hannahhoward — APPROVED** [review]
LGTM, part 2, let's merge it!

**Pull request merged**